File: java/colecciones/Ejemplo02/com/ditamex/main/ArrayListTest.java
package com.ditamex.main;

import java.util.ArrayList;

public class ArrayListTest
{
    /*
     * Revisar la documentación de la clase ArrayList
     * https://docs.oracle.com/javase/8/docs/api/java/util/ArrayList.html
     */

    public static void main(String args[]) {
        
        //Crear una instancia de ArrayList con "cero" elementos
        ArrayList colores = new ArrayList();
        
        //Se agregan valores iniciales al vector
        colores.add("Rojo");
        colores.add("Blanco");
        colores.add("Amarillo");
        colores.add("Negro");
        colores.add("Azul"); 
        
        //Explicar el método "toString" de los vectores
        System.out.println("Colores: " + colores);        
        
        //Primer operación "add"
        colores.add("Rosa");
        System.out.println("Colores: " + colores);

        //Segunda operación "set"
        colores.set(3, "Verde");
        System.out.println("Colores: " + colores);        
                
        //Tercera operación "remove"
        colores.remove(1);
        System.out.println("Colores: " + colores);  
        
        //Cuarta operación "get"
        String color = (String) colores.get(4);
        System.out.println("Color: " + color);
    }
}

File: c++/1.4 Iteradores y Colecciones de Objetos/1.4.3 Colecciones/Ejemplo02/ArrayListTest.cpp
#include <iostream>
#include <list>

using namespace std;

void imprimeLista(string nombre, list<string> miLista) {
  cout << nombre << ": [";
  for (list<string>::iterator it = miLista.begin(); it != miLista.end();) {
    cout << *it;
    if (++it != miLista.end()) {
      cout << ", ";
    }
  }
  cout << "]" << endl;
}

int main(int argc, char const *argv[]) {
  list<string> colores;

  // Se agregan valores iniciales al vector
  colores.push_back("Rojo");
  colores.push_back("Blanco");
  colores.push_back("Amarillo");
  colores.push_back("Negro");
  colores.push_back("Azul");

  // imprime Lista
  imprimeLista("Colores", colores);

  // Primer operación "add: push_back()"
  colores.push_back("Rosa");
  imprimeLista("Colores", colores);

  // Segunda operación "set: next()"
  *next(colores.begin(), 3) = "Verde";
  imprimeLista("Colores", colores);

  // Tercera operación "remove: erase()"
  colores.erase(next(colores.begin(), 1));
  imprimeLista("Colores", colores);

  // Cuarta operación "get: next()"
  string color = *next(colores.begin(), 4);
  cout << "Color: " << color << endl;

  return 0;
}